src/components/ProductTours/AdminOnboardingTours/AdminOnboardingTours.tsx

```diff
@@ -46,6 +46,14 @@ const AdminOnboardingTours: FC<AdminOnboardingToursProps> = ({
     adminUuid, aiButtonVisible, currentStep, setCurrentStep, enterpriseSlug, onClose, targetSelector,
   });
 
+  // Reset step for use case where we need to navigate to a different page but still
+  // on the same flow
+  useEffect(() => {
+    if (targetSelector === 'subscription-plans-detail-page') {
```

Comment:
Would it make sense to add another field to the TourStep object, indicating that we should reset the tour steps?  When we have more steps in the future that require resetting, keying off their `targetSelector` will become an unwieldy solution.